omod/src/main/java/org/openmrs/module/coreapps/page/controller/clinicianfacing/PatientPageController.java

```diff
@@ -140,6 +143,8 @@ else if (patient.isVoided() || patient.isPersonVoided()) {
 
         model.addAttribute("baseDashboardUrl", coreAppsProperties.getDashboardUrl());  // used for breadcrumbs to link back to the base dashboard in the case when this is used to render a context-specific dashboard
         model.addAttribute("dashboard", dashboard);
+        
+        model.addAttribute("extraNamePersonAttrs", getPersonNamePersonAttrs(patient));
```

Comment:
@Ruhanga we want the breadcrumb to be configurable to display _any_ collection of person attributes based on a setting. Not just those that represent names. What if we want to do something like this?

<img src="https://user-images.githubusercontent.com/12812196/68600447-d734db80-04a2-11ea-8407-43d36b9c40b5.png" data-canonical-src="https://user-images.githubusercontent.com/12812196/68600447-d734db80-04a2-11ea-8407-43d36b9c40b5.png" width="300" />

Your code has to be more generic to allow such use cases.

Reply:
Oh yes @mks-d. This is simply demo code from the name use-case. I'm making this generic. Thank you 👍.
So something like
`model.addAttribute("breadCrumbDetails", getBreadCrumbDetails(patient));`
may do? Any suggestions?

Reply:
Yes good, but plural: "breadcrumbs details".

It would be good if the entry string (`"("` in my example), the separation string (`", "` in my example) and the final string (`")"` in my example) could be configured.

Try this ;-)

```java
// The list of person attr. values to include
List<String> attrs = Arrays.asList("مُحَمَّد‎", "Lebanese");

// Setting up the template string
String templateString = "";
for (int i = 1; i <= attrs.size(); i++) {
  templateString += "${attr" + i + "}" + "${sep}";
}
templateString = StringUtils.chomp(templateString, "${sep}");
templateString = "${prefix}" + templateString + "${suffix}";


Map<String, String> valuesMap = new HashMap<String, String>();

// Configuring the prefix, suffix and sep.
valuesMap.put("prefix", "(");
valuesMap.put("suffix", ")");
valuesMap.put("sep", ", ");

// Mapping the attrs. values
for (int i = 1; i <= attrs.size(); i++) {
  valuesMap.put("attr" + i, attrs.get(i - 1));
}

// Subsituting the breadcrumbs details
StringSubstitutor sub = new StringSubstitutor(valuesMap);
String resolvedString = sub.replace(templateString);
```

Reply:
So I let groovy handle the formatting as seen here 422d8a0 .